docs/advanced_topics/images/amp.rst

```diff
@@ -0,0 +1,282 @@
+
+ - ``activate_amp_mode`` is a context manager which can be invoked using Python's
+   ``with`` syntax. In the body of the ``with`` statement, AMP mode would be active.
+
+ - ``amp_mode_active`` is a function that returns ``True`` when AMP mode is active.
+
+Next, we need to define a view that wraps Wagtail's builtin ``serve`` view and
+invokes the ``activate_amp_mode`` context manager:
+
+.. code-block:: python
+
+    # <app>/amp_views.py
+
+    from django.template.response import SimpleTemplateResponse
+    from wagtail.core.views import serve as wagtail_serve
+
+    from .amp_utils import activate_amp_mode
+
+    def serve(request, path):
+        with activate_amp_mode():
+            response = wagtail_serve(request, path)
+
+            # Render template responses now while AMP mode is still active
+            if isinstance(response, SimpleTemplateResponse):
+                response.render()
+
+            return response
+
+Then we need to create a ``amp_urls.py`` file in the same app:
+
+.. code-block:: python
+
+    # <app>/amp_urls.py
+
+    from django.conf.urls import url
+    from wagtail.core.urls import serve_pattern
+
+    from . import amp_views
+
+    urlpatterns = [
+        url(serve_pattern, amp_views.serve, name='wagtail_amp_serve')
+    ]
+
+Finally, we need to update the project's main ``urls.py`` to use this new URLs
+file for the ``/amp`` prefix:
+
+.. code-block:: python
+
+    # <project>/urls.py
+
+    from myapp import amp_urls as wagtail_amp_urls
+
+    urlpatterns += [
+        # Change this line to point at your amp_urls instead of Wagtail's urls
+        url(r'amp/', include(wagtail_amp_urls)),
+
+        url(r'', include(wagtail_urls)),
+    ]
+
+After this, there shouldn't be any noticable difference to the AMP version of
+the site.
+
+Write a template context process so that AMP state can be checked in templates
+------------------------------------------------------------------------------
+
+This is optional, but worth doing so we can confirm that everything is working
+so far.
+
+Add a ``amp_context_processors.py`` file into your app that contains the
+following:
+
+.. code-block:: python
+
+    # <app>/amp_context_processors.py
+
+    from .amp_utils import amp_mode_active
+
+    def amp(request):
+        return {
+            'amp_mode_active': amp_mode_active(),
+        }
+
+Now add the path to this context processor to the
+``['OPTIONS']['context_processors']`` key of the ``TEMPLATES`` setting:
+
+.. code-block:: python
+
+    # Either <project>/settings.py or <project>/settings/base.py
+
+    TEMPLATES = [
+        {
+            ...
+
+            'OPTIONS': {
+                'context_processors': [
+                    ...
+                    # Add this after other context processors
+                'myapp.amp_context_processors.amp',
+                ],
+            },
+        },
+    ]
+
+You should now be able to use the ``amp_mode_active`` variable in templates.
```

Comment:
Does this need an example on how to vary your template based on the amp mode?

Reply:
This is covered below, did you mean add a more realistic example?

Reply:
Yes.
My other thought was that lots of people would use separate templates all together. And that could be either includes in the template, or vary the template in the view/ serve method. But as I write it I realize that is one of those things that we should not be prescriptive about and the ` {% if amp_mode_active %}{% endif %}` example below is good enough

Reply:
Added a section about this. I've also moved it to the "Advanced topics" root as it's no longer specific to images.